app/src/main/java/org/mozilla/fenix/settings/RadioButtonPreference.kt

```diff
@@ -35,6 +37,10 @@ class RadioButtonPreference : Preference {
         radioGroups.add(radioPreference)
     }
 
+    fun onClickListener(listener: (() -> Unit)) {
```

Comment:
I don't believe this current usage would cause any problems, but indulge me. I think we should perhaps nullify the clickListener when the preference is unbound to eliminate the possibility of a future developer keeping a strong reference to anything we don't want kept, like Context.

Reply:
To clarify the possible issue, could you give an example where a memory leak can happen because of the `clickListener` reference?

Reply:
On second thought, I think you're probably safe. The issue lies with passing Context or the Fragment to inner classes or to process-bound classes that can outlive the fragment or activity. Here's the well-spoken Mark Murphy on the subject. I think in this case, the Preference is likely cleaned up. I'm being overly-cautious is all. https://stackoverflow.com/a/5010949/1412973

Reply:
Thanks for the reference, I'll take a look at it!